dvc/remote/base.py

```diff
@@ -288,16 +294,71 @@ def changed_cache(self, checksum):
 
         return False
 
+    def _parallel(self, func, args):
+        """Wraps iterator into threadsafe_iterator and executes func in JOBS
+        threads in parallel
+        """
+        results = []
+        args = threadsafe_iterator(args)
+        with ThreadPoolExecutor(max_workers=self.JOBS) as executor:
+            futures = [executor.submit(func, args) for i in range(self.JOBS)]
```

Comment:
I'm trying to understand this `threadsafe_iterator` and the way it's used with `ThreadPoolExecutor`. Is it to save some memory? It feels like something like queue could be used or we just can submit individual items, no?

Reply:
Or is it just because of that progess loop wrapper? Btw, could we just reuse the progress_callback mechanism here?

Reply:
> It feels like something like queue could be used or we just can submit individual items, no?
Yes, it transforms the iterator into the queue, exactly.